src/users/user.php

```diff
@@ -112,37 +78,67 @@ public function post()
 			}
 			else
 			{
-				if ($fidentifier)
-				{
-					$user = new JUser;
+				ApiError::raiseError(400, JText::_('PLG_API_USERS_USER_NOT_FOUND_MESSAGE'));
 
-					if ($formData['password'] == '')
-					{
-						ApiError::raiseError(400, JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE'));
+				return;
+			}
+		}
+		// Check if $userIdentifier is not set - POST / CREATE user case
+		else
+		{
+			if (!$iAmSuperAdmin)
+			{
+				ApiError::raiseError(400, JText::_('JERROR_ALERTNOAUTHOR'));
 
-						return;
-					}
+				return;
+			}
 
-					// Set default group if nothing is passed for group.
-					if (empty($formData['groups']))
-					{
-						$formData['groups'] = array($params->get("new_usertype", 2));
-					}
+			// Validate required fields
+			if ($formData['username'] == '' || $formData['name'] == '' || $formData['email'] == '')
+			{
+				ApiError::raiseError(400, JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE'));
 
-					// Create new user.
-					$response = $this->storeUser($user, $formData, 1);
-					$this->plugin->setResponse($response);
+				return;
+			}
 
-					return;
-				}
-				else
-				{
-					ApiError::raiseError(400, JText::_('PLG_API_USERS_USER_ABSENT_MESSAGE'));
+			// Set default group if nothing is passed for group.
+			if (empty($formData['groups']))
+			{
+				$formData['groups'] = array($params->get("new_usertype", 2));
+			}
 
-					return;
-				}
+			// Get a blank user object
+			$user = new JUser;
+
+			// Create new user.
+			$response = $this->storeUser($user, $formData, 1);
+			$this->plugin->setResponse($response);
+
+			return;
+		}
+	}
+
+	/**
+	 * Funtion to remove sensitive user info fields like password
+	 *
+	 * @param   Object  $user    The user object.
+	 * @param   Array   $fields  Array of fields to be unset
+	 *
+	 * @return  object|void  $user
+	 *
+	 * @since   2.0.1
+	 */
+	protected function sanitizeUserFields($user, $fields = array('password', 'password_clear', 'otpKey', 'otep'))
```

Comment:
The second argument should be a class constant.

Reply:
Instead of returning I suggest change the method to accept the variable by reference ?